src/exercises/Exercises.js

```diff
@@ -45,12 +61,83 @@ export default function Exercises({ api, articleID }) {
   }, []);
 
   function initializeExercises(bookmarks, title) {
-    setbookmarksToStudyList(bookmarks);
     NUMBER_OF_EXERCISES = bookmarks.length;
-    setCurrentBookmarkToStudy(bookmarks[currentIndex]);
+    setSession(bookmarks);
     setTitle(title);
   }
 
+  function setSession(bookmarks) {
+    let bookmarkSum = EXERCISES.reduce((a, b) => a + b.requiredBookmarks, 0);
+    let batches = parseInt(NUMBER_OF_EXERCISES / bookmarkSum);
+    let rest = NUMBER_OF_EXERCISES % bookmarkSum;
+    let exercises = [];
+    if (NUMBER_OF_EXERCISES < 7) {
+      let bookmarks = NUMBER_OF_EXERCISES;
+      while (bookmarks > 0) {
+        for (let i = EXERCISES.length - 1; i > 0; i--) {
+          if (bookmarks === 0) break;
+          let exercise = {
+            type: EXERCISES[i].type,
+            requiredBookmarks: EXERCISES[i].requiredBookmarks,
+            bookmarks: [],
+          };
+          exercises.push(exercise);
+          bookmarks--;
+        }
+      }
+    } else {
+      for (let i = 0; i < batches; i++) {
+        for (let j = EXERCISES.length - 1; j >= 0; j--) {
+          let exercise = {
+            type: EXERCISES[j].type,
+            requiredBookmarks: EXERCISES[j].requiredBookmarks,
+            bookmarks: [],
+          };
+          exercises.push(exercise);
+        }
+      }
+      while (rest > 0) {
+        for (let k = EXERCISES.length - 1; k >= 0; k--) {
+          if (rest >= EXERCISES[k].requiredBookmarks) {
+            let exercise = {
+              type: EXERCISES[k].type,
+              requiredBookmarks: EXERCISES[k].requiredBookmarks,
+              bookmarks: [],
+            };
+            exercises.push(exercise);
+            rest--;
+          }
+        }
+      }
+    }
+    setExerciseSession(exercises);
+    setBookmarksByExercise(bookmarks, exercises);
+  }
+
+  function setBookmarksByExercise(bookmarkList, session) {
+    let k = 0;
```

Comment:
method comment could help with explaining what this is about.

Reply:
Done. See commit fca2e0c148b12bc6f78c8591952c1d99ac95ddfd